File: src/Wrappers/AntRunner.Wrapper.Python/lib/AntWrapper.py
import sys, inspect, json
import AntRunner.Interface

if len(sys.argv) < 2 or not sys.argv[1]:
	raise Exception("Must supply Ant module name")

if len(sys.argv) > 2 and sys.argv[2].startswith("debug:"):
	port = int(sys.argv[2][6:])
	if port > 1000:
		import ptvsd
		ptvsd.enable_attach(address=('127.0.0.1', port), redirect_output=False)
		ptvsd.wait_for_attach()

LoadingAnt = __import__(sys.argv[1])

def MakeAnt(mod):
    for name, obj in inspect.getmembers(mod):
        if inspect.isclass(obj) and issubclass(obj, AntRunner.Interface.Ant) and obj is not AntRunner.Interface.Ant:
            return obj()

def DecodeGameState(s):
	d = json.loads(s)
	if d["Response"] is not None:
		d["Response"]=AntRunner.Interface.EchoResponse(**d["Response"])
	return  AntRunner.Interface.GameState(**d)


Ant = MakeAnt(LoadingAnt)


for line in sys.stdin:
	cmd = line[:1]
	if cmd == "T":
		s = DecodeGameState(line[1:])
		Ant.Tick(s)
		a = Ant.Action
		sys.stdout.write(f"{a}\n")
		Ant.Action = 0
	elif cmd == "I":
		i = json.loads(line[1:])
		Ant.Initialize(i[0],i[1],i[2],i[3],i[4])
	elif cmd == "N":
		sys.stdout.write(f"{Ant.Name}\n")
	elif cmd == "P":
		sys.stdout.write("Ping\n")

	sys.stdout.flush()

File: src/Wrappers/AntRunner.Wrapper.Ruby/lib/AntWrapper.rb
if !File.file?(ARGV[0])
	raise("Ant file '#{ARGV[0]}' does not exist!")
end
require("json")
require_relative("AntRunner.rb")
require(ARGV[0])

STDOUT.sync = true

__CurrentAnt__ = AntRunner::Interface::Ant.Descendants().first.new()
__Running__ = true

while __Running__ && STDIN.gets
	begin
		line = $_.strip!
		if line.nil? || line.length == 0
			next
		end
		cmd = line[0]
		case cmd
		when "T"
			__CurrentAnt__.Tick(AntRunner::Interface::GameState.from_json(line[1..-1]))
			STDOUT.write("ok\n")
		when "A"
			a = __CurrentAnt__.Action
			STDOUT.write("#{a}\n")
			__CurrentAnt__.Action = AntRunner::Interface::AntAction::Wait
		when "I"
			i = JSON.parse(line[1..-1])
			__CurrentAnt__.Initialize(i[0], i[1], i[2], i[3], i[4])
		when "N"
			STDOUT.write("#{__CurrentAnt__.Name}\n")
		when "P"
			STDOUT.write("Ping\n")
		when "X"
			__Running__ = false
		else
			STDOUT.write("Invalid command\n")
		end
	rescue
		#Do Nothing
	end
end